src/components/Card/components/Header/Header.tsx

```diff
@@ -13,86 +13,85 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from '@emotion/styled';
+import React, { FC, ReactNode, MouseEvent, KeyboardEvent } from 'react';
 import { css } from '@emotion/core';
+import { Dispatch as TrackingProps } from '@sumup/collector';
 
-import CloseButton from '../../../CloseButton';
-import { childrenPropType } from '../../../../util/shared-prop-types';
+import styled, { StyleProps } from '../../../../styles/styled';
+import {
+  CloseButton,
+  CloseButtonProps,
+} from '../../../CloseButton/CloseButton';
 
-const baseStyles = ({ theme }) => css`
+export interface CardHeaderProps {
+  /**
+   * Heading to be shown.
+   */
+  children: ReactNode;
+  /**
+   * Text label for the close button for screen readers.
+   * Important for accessibility.
+   */
+  labelCloseButton?: string;
+  /**
+   * Additional data that is dispatched with the tracking event.
+   */
+  tracking?: TrackingProps;
+  /**
+   * Callback for the close button. If not specified, the button won't
+   * be shown.
+   */
+  onClose?: (event: MouseEvent | KeyboardEvent) => void;
+}
+
+const baseStyles = ({ theme }: StyleProps) => css`
   label: card__header;
   align-items: center;
   display: flex;
   justify-content: space-between;
   margin-bottom: ${theme.spacings.giga};
 `;
 
-const noHeadingStyles = ({ children }) =>
-  !children[0] &&
-  css`
-    label: card__header--no-heading;
-    justify-content: flex-end;
-  `;
+const CardHeaderContainer = styled('header')<CardHeaderProps>(baseStyles);
 
-const CardHeaderContainer = styled('header')(baseStyles, noHeadingStyles);
-
-const closeButtonStyles = ({ theme }) => css`
+const closeButtonStyles = ({ theme }: StyleProps) => css`
   margin-top: -${theme.spacings.byte};
   margin-right: -${theme.spacings.kilo};
   margin-bottom: -${theme.spacings.byte};
 `;
 
-const CardHeaderCloseButton = styled(CloseButton)(closeButtonStyles);
+const CardHeaderCloseButton = styled(CloseButton)<CloseButtonProps>(
+  closeButtonStyles,
+);
 
 /**
  * Header used in the Card component. Used for styling and alignment
  * purposes only.
  */
-const CardHeader = ({
-  onClose,
+export const CardHeader: FC<CardHeaderProps> = ({
   children,
+  tracking,
+  onClose,
   labelCloseButton,
-  tracking = {},
   ...props
 }) => (
-  <CardHeaderContainer {...props}>
+  <CardHeaderContainer
+    labelCloseButton={labelCloseButton}
+    tracking={tracking}
+    onClose={onClose}
```

Comment:
Why did you add these props to this element?

Reply:
🤔 
I was trying to pass the attributes ahead .. so I included them to the `CardHeaderContainer`, to the `CardHeaderContainer` pass the attribute instead of just use directly int the `CardHeaderCloseButton`